frontend/src/metabase/query_builder/components/expressions/ExpressionEditorSuggestions/ExpressionEditorSuggestions.tsx

```diff
@@ -289,45 +305,42 @@ function ExpressionEditorSuggestionsListItem({
 function Footer({
   suggestion,
   highlightedIndex,
+  onHighlightSuggestion,
 }: {
   suggestion: WithIndex<SuggestionFooter>;
   highlightedIndex: number;
+  onHighlightSuggestion: (index: number) => void;
 }) {
   function handleMouseDownCapture(evt: MouseEvent) {
     // prevent the dropdown from closing
     evt.preventDefault();
   }
 
+  const handleMouseMove = useCallback(() => {
+    onHighlightSuggestion(suggestion.index);
```

Comment:
It might be good to call `onHighlightSuggestion` only when `suggestion.index` is different than `highlightedIndex`.